composer/trainer/trainer.py

```
@@ -1209,31 +1230,61 @@ def _get_autoresume_checkpoint(
-                    pass
-        # Require all ranks to have local checkpoint if we wish to restore from it
-        latest_checkpoint_exists = self._device.tensor_to_device(
-            torch.tensor([os.path.exists(latest_checkpoint_path)], dtype=torch.uint8))
-        dist.all_reduce(latest_checkpoint_exists, reduce_operation='MIN')
-        # If latest checkpoint is saved locally, change load_path to it
-        if int(latest_checkpoint_exists.item()) == 1:
-            return latest_checkpoint_path
+        if not os.path.exists(latest_checkpoint_path) and (dist.get_global_rank() == 0 or self.deepspeed_enabled):
+            self._attempt_checkpoint_download(save_folder, latest_checkpoint_path, save_latest_remote_file_name,
+                                              loggers, load_progress_bar)
+
+        if self.deepspeed_enabled:
+            # Require all ranks to have their own local checkpoint if we wish to restore from it for deepspeed
+            latest_checkpoint_exists_on_all_ranks = self._device.tensor_to_device(
+                torch.tensor([os.path.exists(latest_checkpoint_path)], dtype=torch.uint8))
+            dist.all_reduce(latest_checkpoint_exists_on_all_ranks, reduce_operation='MIN')
+
+            if int(latest_checkpoint_exists_on_all_ranks.item()) == 0:
+                raise RuntimeError('DeepSpeed was enabled, but checkpoints were not found for all ranks')
+            else:
+                return latest_checkpoint_path
+        else:
+            # The checkpoint must at least exist for rank zero
+            latest_checkpoint_exists_on_rank_zero = self._device.tensor_to_device(
+                torch.tensor([os.path.exists(latest_checkpoint_path) and dist.get_global_rank() == 0],
+                             dtype=torch.uint8))
+            dist.all_reduce(latest_checkpoint_exists_on_rank_zero, reduce_operation='MAX')
+
+            if int(latest_checkpoint_exists_on_rank_zero.item()) == 0:
+                raise RuntimeError(
```

I think we should return `None` here (cc: @mvpatel2000 ?) so that a fresh run with `autoresume: True` still works.

---

I did have log an info and return `None` at one point. That results in silent failure if the checkpoint doesn't exist/can't be found for some reason. I wasn't sure which was worse, silent failure in the case of it being missing? or erroring on the first run with autoresume true?

---

I think log a warning and return None makes sense

---

Later on in the codepath, we already log a warning, so i think returning None makes sense.

THe idea here is that the same code/settings while automatically pick up on old checkpoints and resume.

---

Ah ok, makes sense.